### PR TITLE
Update Node and dependencies

### PR DESCRIPTION
**Pull request by niallcolfer**
### Description
Updating Node.Js to a current LTS version (Node.Js 10), and dependencies to supported versions to compile with security introduced by out of date dependencies. 

### Risk
High: upgrade Node.Js jump 6 major release from Node.Js 4 to Node.Js 10 is always risky.

@zendesk/radar @zendesk/argonauts

### CODE REVIEW

**Comment by sebdeckers** [comment]
FWIW, I tried bumping `mocha` yesterday and found that `--ui exports` caused this `ReferenceError: describe is not defined` error. Might want to refactor the tests to not rely on https://github.com/osher/mocha-ui-exports anymore.

**Comment by thomasbui93** [comment]
I tried to list out vulnerable packages with `npm audit` and updated them as much as possible. The remain vulnerable packages are `ws`, `engine.io-client`, `parsejson`, `debug`, these are belonged to radar_client (https://github.com/zendesk/radar_client).
I will proceed to update radar_client and link the pull request here.

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "author": "Zendesk, Inc.",
   "license": "Apache-2.0",
   "engines": {
-    "node": ">=4"
+    "node": ">=12"
```
Comment (abmartinr) [inline package.json]: We should bump this as a Major and make sure we add our changes into changelog.md

Otherwise LGTM

Comment (thomasbui93) [reply package.json]: 👍 Yes, this should be >=10 instead too.

**Pull request merged**